### ISSUE TITLE
`DataFrame select:` fails when no rows are selected

### ISSUE THREAD

**Issue by AtharvaKhare** [PolyMathOrg/DataFrame#91]
Running the following block:
```
df := DataFrame withRows: #((1) (2) (3) (4)).
df select: [ :row | (row at: 1) >= 5 ].
```
leads to `SubscriptOutOfBounds: 1`, due to `first` in `DataFrameInternal withRows:`:
```
numberOfColumns := anArrayOfArrays first size.
```

Shouldn't it return an empty Dataframe instead?

**Comment by olekscode** [comment]
Yes, that's definitely a bug!
I think, we can fix it by making sure that `DataFrame >> withRows:` and all related messages return an empty data frame when given an empty array of rows.
```Smalltalk
DataFrame withRows: #(). "a DataFrame(0@0)"
```

**Comment by AtharvaKhare** [comment]
Same is true for `reject`:
```
df := DataFrame withRows: #((1) (2) (3) (4)).
df reject: [ :row | (row at: 1) <= 5 ].
```

**Comment by AtharvaKhare** [comment]
Can I take this up and adding related tests?

**Comment by olekscode** [comment]
I have created a related issue #92 for it

**Comment by olekscode** [comment]
But don't work on this issue now, because I want to plan and organize all tasks and issues first.
We can discuss it tomorrow evening.

**Comment by AtharvaKhare** [comment]
Fixed with #100.
